rules/configuration.smk

```diff
@@ -41,39 +44,49 @@ assert sorted(config['fqext'])[0] == config['fqext1']
 
 # check if a sample is single-end or paired end, and store it
 logger.info("Checking if samples are single-end or paired-end...")
-def get_layout(sample):
-    """ sends a request to ncbi checking whether a sample is single-end or paired-end """
-    api_key = config.get('ncbi_key', "")
-    if api_key is not "":
-        api_key = f'-api_key {api_key}'
-
-    return sample, subprocess.check_output(
-        f'''esearch {api_key} -db sra -query {sample} | efetch {api_key} | grep -Po "(?<=<LIBRARY_LAYOUT><)[^/><]*"''',
-        shell=True).decode('ascii').rstrip()
-
-
-results = []
-tp = ThreadPool(config['ncbi_requests'] // 2)
-config['layout'] = {}
-
-# now do a request for each sample
-for sample in samples.index:
-    if os.path.exists(expand(f'{{result_dir}}/{{fastq_dir}}/{sample}.{{fqsuffix}}.gz', **config)[0]):
-        config['layout'][sample] ='SINGLE'
-    elif all(os.path.exists(path) for path in expand(f'{{result_dir}}/{{fastq_dir}}/{sample}_{{fqext}}.{{fqsuffix}}.gz', **config)):
-        config['layout'][sample] ='PAIRED'
-    else:
-        results.append(tp.apply_async(get_layout, (sample,)))
-        # sleep 1.25 times the minimum required sleep time
-        time.sleep(1.25 / (config['ncbi_requests'] // 2))
-
-# now parse the output and store in config
-config['layout'] = {**config['layout'], **{r.get()[0]: r.get()[1] for r in results}}
+# layout file with a unique name depending on the samples contained
+identifier = str(blake2b(''.join(samples.sort_index().index).encode(), digest_size=12).hexdigest())
+layout_file = './.snakemake/layouts_' + identifier
```

Comment:
So if I first try with samples:
GSM12345
GSM67890

And then again only with sample
GSM12345 

It needs to do the lookup again?